specs/AddHostObjectToScriptWithOrigins.md

```diff
@@ -0,0 +1,282 @@
+By providing event handler for frame created event,
+user can obtain the iframe and add host objects to it specifying the list of origins
+using `AddHostObjectToScriptWithOrigins` method.
+When script in the iframe tries to access a host object the origin of the iframe will be
+checked against the provided list. The Frame object has a Name property with the value of the
+name attribute from the iframe html tag declaring it. The NameChanged event is raised when
+corresponding iframe changes its window.name property.
+
+The host object will be removed during iframe deletion automatically.
+
+
+# Examples
+## Frame created event, frame name changed event, adding and removing host object
+
+## .NET, WinRT
+```c#
+void SubscribeToFrameCreated()
+{
+    webView.CoreWebView2.FrameCreated += delegate (object sender, CoreWebView2FrameCreatedEventArgs args)
+    {
+        try
+        {
+            if (args.Frame.Name.Equals("iframe_name"))
+            {
+                String[] origins = new String[] { "https://appassets.example" };
+                args.Frame.AddHostObjectToScriptWithOrigins("bridge", new BridgeAddRemoteObject(), origins);
+            }
+            args.Frame.NameChanged += delegate (object nameChangedSender, object nameChangedArgs)
+            {
+                CoreWebView2Frame frame = (CoreWebView2Frame)nameChangedSender;
+                if (!frame.Name.Equals("iframe_name"))
+                {
+                    frame.RemoveHostObjectFromScript("bridge");
+                }
+            };
+        }
+        catch (NotSupportedException exception)
+        {
+            MessageBox.Show("Frame.AddHostObjectToScriptWithOrigins failed: " + exception.Message);
+        }
+    };
+}
+```
+
+## Win32 C++
+```cpp
+void SampleClass::SampleMethod()
+{
+    // Register a handler for the FrameCreated event.
+    // This handler can be used to add host objects to the created iframe.
+    CHECK_FAILURE(m_webView->add_FrameCreated(
+        Callback<ICoreWebView2FrameCreatedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2FrameCreatedEventArgs* args) -> HRESULT
+    {
+        wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+        CHECK_FAILURE(args->get_Frame(&webviewFrame));
+
+        // Subscribe to frame name changed event and remove host object when name is changed
+        webviewFrame->add_NameChanged(
+        Callback<ICoreWebView2FrameNameChangedEventHandler>(
+            [this](ICoreWebView2Frame* sender,
+                IUnknown* args) -> HRESULT {
+            LPWSTR newName;
+            CHECK_FAILURE(sender->get_Name(&newName));
+            if (std::wcscmp(newName, L"iframe_name") != 0)
+            {
+                sender->RemoveHostObjectFromScript(L"sample");
+            }
+            return S_OK;
+        }).Get(), NULL);
+
+        LPWSTR name;
+        CHECK_FAILURE(webviewFrame->get_Name(&name));
+        if (!std::wcscmp(name, L"iframe_name"))
```

Comment:
Can we do an explicit comparison to 0 to match the explicit != 0 above and since wcscmp isn't boolean?
```suggestion
        if (std::wcscmp(name, L"iframe_name") == 0)
```